src/utils/traffic-mesh.js

```diff
@@ -57,6 +59,7 @@ const startForwardProxy = async ({ port, frameworkPort, functionsPort, publishDi
     EDGE_HANDLERS_BUNDLER_CLI_PATH,
     '--log-file',
     getPathInProject(['logs', 'traffic-mesh.log']),
+    '--progress',
```

Comment:
Quick question: do we need to spawn this is a new process, as opposed to calling it programmatically? Assuming this is a Node.js process. This would remove the need to parse `stderr`, as information could passed directly using callbacks for example.

Reply:
This is calling a Rust process.

Reply:
I am wondering if terminal output might be a good channel for IPC. What are your thoughts on some alternatives like using the filesystem (e.g. watching for a file), or using a small TCP server?

Reply:
Using pipes for IPC is actually a common choice. You _do_ often set up dedicates pipes for passing the IPC messages though, and don't use stdout / stderr (though I've also seen stdout being used). One prominent example: https://github.com/xi-editor/xi-editor

Pipes have the advantage of being being simple to set up (one syscall), they are automatically connected, they are available on all operating systems, they don't fail and they are automatically cleaned up when you're done with them. Tempfiles and TCP sockets both are either more complex to set up and synchronize or take effort for cleanup. The multiple-connection machine-to-machine IPC case is generally also better served with UDS, but that doesn't exist on Windows other than a super recent insiders build.

Signals might be worth an investigation, but I imagine it's going to be really hard passing actually usable information across them except for "build started", "build complete", and "build failed". This approach can ship detailed error information across process boundaries.

Reply:
Those are all very good points. :+1:

Reply:
I am not sure but I think nodejs also uses pipes if you use their IPC implementation. At least, that would make a lot of sense. 😄

Reply:
> I am not sure but I think nodejs also uses pipes if you use their IPC implementation. 

Yes, [it does](https://github.com/nodejs/node/blob/master/lib/internal/child_process.js#L985), with [JSON or v8-serialized payloads](https://github.com/nodejs/node/blob/master/lib/internal/child_process/serialization.js) depending on the `serialization` option.